### DISCUSSION TITLE
Is it possible to run Quarto on server?

### DISCUSSION THREAD

**Discussion by stanislav-kukucka** [quarto-dev/quarto-cli#9120] [Q&A]
### Description

I would like to run ipynb as a Python script and generate a PDF file with the help of Quarto on the server. Saved on the server. I see just desktop/localhost installation. Is it possible to run Quarto on the server as a headless solution? I found just this guide here [https://quarto.org/docs/computations/python.html](https://quarto.org/docs/computations/python.html)

**Comment by mcanouil** [comment]
I am not sure to follow and understand what you mean by "server" or what you want to do exactly.
Quarto is a CLI.
There is no "desktop" / "server" versions.

Install it wherever you need it.
For instance, in continuous integration on GitHub Actions, or Google Cloud Build.

**Reply by stanislav-kukucka** [reply]
@mcanouil I am thinking about setting a project the way I build analysis in ipynb then ipynb convert to python script (cos I want to dynamically insert value as analyzed website) Question is if the python script runs on the server can communicate with Quarto and save the output as a pdf file in proper layout?

**Reply by mcanouil** [reply]
For clarity and see if I understand:
1. Local: Run whatever using a Jupyter Notebook
2. Local: convert the notebook to a Python script
3. Local/remote: copy the Python script over to the remote ("server")
4. Remote: Run Quarto

Is it your workflow?

If so:
1. Why not directly use the Notebook (with computation) in Quarto?
2. If you intend to run the Python script, when rendering, then look at pre-render script.

Note that "server", or local/remotes makes absolutely no difference or even is irrelevant.
As I said, Quarto is a CLI. Install and use wherever you want.

> Question is if the python script runs on the server can communicate with Quarto and save the output as a pdf file in proper layout?

Still unclear what this means.
A Python script is a Python script.
What do you mean by "communicate"? Calling Quarto from Python? Something else?

What does "save the output as a PDF file in proper layout"?
Quarto renders "qmd" or "ipynb" to a format defined either in the front matter as "metadata", from a project level configuration file "_quarto.yml" or from a parameter using Quarto CLI `quarto render myfile.ipynb --to pdf`.
"Proper layout"? You define that in the same way as the output format.

Here I would suggest to read or read again the Get Started tutorials in https://quarto.org/ to understand the basics of what Quarto does.
